README.md

```diff
@@ -12,8 +12,15 @@ For performance benchmarks of these hash functions and their comparison to other
 ## Merkle
 [Merkle module](./src/merkle/) provides a set of data structures related to Merkle trees. All these data structures are implemented using the RPO hash function described above. The data structures are:
 
-* `MerkleTree`: a regular fully-balanced binary Merkle tree. The depth of this tree can be at most 64.
+* `MerkleError`: a wrapper for the possible error variants when performing Merkle tree operations, including storage failure.
+* `MerklePath`: a Merkle path container, composed of a sequence of nodes. This wrapper struct will encapsulate the Merkle opening verification.
 * `MerklePathSet`: a collection of Merkle authentication paths all resolving to the same root. The length of the paths can be at most 64.
+* `MerkleTree`: a regular fully-balanced binary Merkle tree. The depth of this tree can be at most 64.
+* `NodeIndex`: a Merkle tree address for it's nodes, consisting of a tuple depth+value.
```

Comment:
nit: this should be "... for its nodes..." Also, I would probably say "...consisting of a tuple containing `depth` and `value`."

Reply:
Fixed!